src/Ui/Table.elm

```diff
@@ -0,0 +1,464 @@
+updateColumn k f config =
+    { config | columns = Dict.update k (Maybe.map f) config.columns }
+
+
+cellContentStyle : Attribute msg
+cellContentStyle =
+    css [ Css.flexGrow (Css.num 1) ]
+
+
+unwrapColumnContent : ColumnContent msg -> Html msg
+unwrapColumnContent (ColumnContent x) =
+    x
+
+
+cellColumns : Flex -> List (ColumnContent msg) -> ColumnContent msg
+cellColumns w =
+    ColumnContent
+        << div
+            [ css
+                ([ Css.displayFlex
+                 , Css.flexDirection Css.row
+                 , Css.height (Css.pct 100)
+                 , Css.whiteSpace Css.normal
+                 ]
+                    ++ flexToStyle w
+                )
+            ]
+        << List.map unwrapColumnContent
+
+
+
+-- Table.cellColumn Table.Grow [ row Table.Grow [ text ..., int... ], icon ..., ]
+
+
+cellText : TextStyle -> String -> ColumnContent msg
+cellText textStyle =
+    ColumnContent << Text.customView [ cellContentStyle ] textStyle
+
+
+cellInt : Int -> ColumnContent msg
+cellInt =
+    ColumnContent << Text.customView [ cellContentStyle ] TextStyle.monospace << String.fromInt
+
+
+cellFloat : Float -> ColumnContent msg
+cellFloat =
+    ColumnContent << Text.customView [ cellContentStyle ] TextStyle.monospace << String.fromFloat
+
+
+cellLink : TextStyle -> { url : String, text : String } -> ColumnContent msg
+cellLink textStyle { url, text } =
+    ColumnContent <| a [ href url ] [ Text.view (textStyle |> TextStyle.withColor TextStyle.linkColor) text ]
+
+
+cellCustom : Html msg -> ColumnContent msg
+cellCustom =
+    ColumnContent
+
+
+type Flex
+    = Auto
+    | Shrink
+    | Grow
+
+
+flexToStyle : Flex -> List Css.Style
+flexToStyle x =
+    case x of
+        Auto ->
+            [ Css.flex <| Css.num 1 ]
+
+        Shrink ->
+            [ Css.flexShrink (Css.num 0), Css.flexBasis Css.auto ]
+
+        Grow ->
+            [ Css.flexGrow (Css.num 1) ]
+
+
+cellRows : Flex -> List (ColumnContent msg) -> ColumnContent msg
+cellRows w =
+    ColumnContent
+        << div
+            [ css
+                ([ Css.displayFlex
+                 , Css.flexDirection Css.column
+                 , Css.whiteSpace Css.normal
+                 ]
+                    ++ flexToStyle w
+                )
+            ]
+        << List.map unwrapColumnContent
+
+
+headerStyle : List Css.Style
+headerStyle =
+    toCssStyle (TextStyle.bodySmall |> TextStyle.withColor Ui.Palette.gray300)
+
+
+cellPadding : Attribute msg
+cellPadding =
+    css [ Css.padding2 (Css.px 10) (Css.px 8) ]
+
+
+alignment : { a | alignment : ColumnAlignment } -> Attribute msg
+alignment col =
+    case col.alignment of
+        Left ->
+            css [ Css.justifyContent Css.left, Css.textAlign Css.left ]
+
+        Center ->
+            css [ Css.justifyContent Css.center, Css.textAlign Css.center ]
+
+        Right ->
+            css [ Css.justifyContent Css.right, Css.textAlign Css.right ]
+
+
+view :
+    List (Attribute msg)
```

Comment:
if this is supposed to be customizable, we will need to be able to assign attributes to a few different parts of the component, no?

Reply:
I don't think that should be necessary, if we see the need we can add the ability to do so later. I rather be strict here to avoid potentially uncessary complexity.